### PR TITLE
Add remaining custom defines

### PR DESCRIPTION
**Pull request by jcpunk**
Not too sure why github closed PR#4 or why I can't seem to get it to re-open....

In any case, here is another stab at the remaining custom defines.

### CODE REVIEW

**Comment by jyaworski** [comment]
Tests fail, but it looks good so far.

**Comment by jcpunk** [comment]
Any ideas how I can fix the test that fails?

**Comment by jyaworski** [comment]
@jcpunk forget the jruby failures. It looks like adding this changed some default to where a file is no longer in the catalog.

File: spec/classes/init_spec.rb [hunk 1]
```diff
@@ -131,4 +131,41 @@
...
+    it { should contain_concat_fragment('squid_header').with_target('/tmp/squid.conf') }
+    it { should contain_concat_fragment('squid_snmp_port_2000').with_content(/^snmp_port\s+2000\s+special for 2000$/) }
+    it { should contain_concat_fragment('squid_snmp_port_2000').with_content(/^if \${process_number} = 3$/) }
+    it { should contain_concat_fragment('squid_snmp_port_2000').with_content(/^endif$/) }
+  end
+
+  context 'with cache_dir parameters set' do
+    let :params do
+      { config: '/tmp/squid.conf',
+        cache_dirs: { '/data,' =>  { 'type'           => 'special',
```
Comment (traylenator) [inline spec/classes/init_spec.rb]: There is a , in there that should not be there, after /data in the key.

**Comment by jcpunk** [comment]
I must have looked at that a dozen times and not seen it.

**Comment by traylenator** [comment]
@jcpunk thanks again, it's all looking good and I have now fixed tests again.

Please rebase, I'll merge and then get a release out  - thanks.

Steve.

**Comment by jcpunk** [comment]
Rebased

**Comment by jyaworski** [comment]
@jcpunk one rubocop error. rubocop -a should fix it.

**Comment by jcpunk** [comment]
Done

**Pull request merged**